src/pages/index.js

```diff
@@ -40,15 +77,88 @@ function createCardElement(data) {
   cardImage.alt = data.name;
   cardTitle.textContent = data.name;
 
-  cardLikedBtn.addEventListener("click", () => cardLikedBtn.classList.toggle("card__like-btn--liked"));
+  if (data.isLiked) {
+    cardLikedBtn.classList.add("card__like-btn--liked");
+  }
+
+  cardLikedBtn.addEventListener("click", () => {
+    const isCurrentlyLiked = cardLikedBtn.classList.contains("card__like-btn--liked");
+
+    api.toggleLike(data._id, !isCurrentlyLiked)
+      .then(updatedCard => {
+        cardLikedBtn.classList.toggle("card__like-btn--liked", updatedCard.isLiked);
+      })
+      .catch(err => console.error("Error toggling like:", err));
+  });
+
   cardImage.addEventListener("click", () => openImageModal(data.link, data.name));
-  deleteBtn.addEventListener("click", (e) => { e.stopPropagation(); cardElement.remove(); });
+
+  deleteBtn.addEventListener("click", (e) => {
+    e.stopPropagation();
+    selectedCard = cardElement;
+    selectedCardId = data._id;
+    openPopup(deleteModal);
+  });
 
   return cardElement;
 }
 
+deleteConfirmBtn.addEventListener("click", (event) => {
+  event.preventDefault();
+  if (selectedCardId) {
+    const originalText = deleteConfirmBtn.textContent;
+    deleteConfirmBtn.textContent = "Deleting...";
+
+    api.deleteCard(selectedCardId)
+      .then(() => {
+        selectedCard.remove();
+        selectedCard = null;
+        selectedCardId = null;
+        closePopup(deleteModal);
+      })
+      .catch(err => console.error("Error deleting card:", err))
+      .finally(() => {
+        deleteConfirmBtn.textContent = originalText;
+      });
+  }
+});
+
+deleteCancelBtn.addEventListener("click", () => {
+  closePopup(deleteModal);
+  selectedCard = null;
+  selectedCardId = null;
+});
+
+avatarEditBtn.addEventListener("click", () => {
+  openPopup(avatarModal);
+  resetValidation(avatarForm, Settings);
+});
+
+function editAvatarFormSubmit(evt) {
+  evt.preventDefault();
+  const avatarUrl = avatarInput.value.trim();
+
+  if (avatarUrl) {
+    const originalText = avatarSubmitBtn.textContent;
+    avatarSubmitBtn.textContent = "Saving...";
```

Comment:
`handleSubmit(makeRequest, evt)` has this code inside itself, so you don't need to repeat it here


Here will be only this:

```js


function editAvatarFormSubmit(evt) {
  const avatarUrl = avatarInput.value.trim();

  if (avatarUrl) {
    function makeRequest() {
      return api.editProfileAvatar({ avatar: avatarUrl }).then((userData) => {
        avatarPreview.src = userData.avatar || avatar;
        closePopup(avatarModal);
      });
    }
    handleSubmit(makeRequest, evt);
  }
}

```